client/scripts/views/components/search_bar.ts

```diff
@@ -225,295 +269,376 @@ const getBalancedContentListing = (unfilteredResults: any[], types: SearchType[]
-        });
-      const tokens = await getTokens();
-      app.searchCache[ALL_RESULTS_KEY]['tokens'] = tokens;
-    } catch (err) {
-      app.searchCache[ALL_RESULTS_KEY]['tokens'] = [];
+
+    if(historyList.length > 0) {
+      historyList.push(
+        m(ListItem, {class: 'search-history-no-results upper-border',
+          label: 'Tip: You can use operators like \'single quotes\', and the keyword "or" to limit your search!'})
+      )
     }
-    app.searchCache[ALL_RESULTS_KEY].loaded = true;
-    m.redraw();
-  }
-};
 
-const emptySearchPreview : m.Component<{ searchTerm: string }, {}> = {
-  view: (vnode) => {
-    const { searchTerm } = vnode.attrs;
-    const message = app.activeId()
-      ? `No results in ${app.activeId()}. Search Commonwealth?`
-      : 'No communities found';
-    return m(ListItem, {
-      class: 'no-results',
-      label: [
-        m('b', searchTerm),
-        m('span', message)
-      ],
-      onclick: (e) => {
-        if (searchTerm.length < 4) {
-          notifyError('Query must be at least 4 characters');
-        }
-        const params = `q=${encodeURIComponent(searchTerm.toString().trim())}`;
-        m.route.set(`/search?${params}`);
-      }
-    });
-  }
-};
+    const scopeTitle = m(ListItem, {class: 'disabled', label: 'Limit search to:'})
 
-export const SearchBar : m.Component<{}, {
-  results: any[],
-  searchTerm: string,
-  errorText: string,
-  focused: boolean,
-  closeResults: Function,
-  hideResults: boolean,
-  inputTimeout: any,
-  isTyping: boolean,
-}> = {
-  view: (vnode) => {
-    if (!vnode.state.searchTerm) vnode.state.searchTerm = '';
+    const scopeToButton = (scope, disabled) => {
+      return m(Button, {
+        size: Size.SM,
+        class: `${disabled ? 'disabled' : ''}`,
+        active: vnode.state.searchQuery.searchScope.includes(scope),
+        onclick: () => {
+          vnode.state.searchQuery.toggleScope(scope)
+          search(vnode.state.searchQuery, vnode.state)
+        },
+        onmouseover: () => {vnode.state.filterMenuActive = true},
+        onmouseout: () => {vnode.state.filterMenuActive = false},
+        label: scope
+      })
+    }
 
-    const { results, searchTerm } = vnode.state;
-    const showDropdownPreview = !m.route.get().includes('/search?q=');
-    const isMobile = (window.innerWidth < 767.98);
-    const LoadingPreview = m(List, {
-      class: 'search-results-loading'
-    }, [ m(ListItem, { label: m(Spinner, { active: true }) }) ]);
-    const searchResults = (!results || results?.length === 0)
-      ? (app.searchCache[searchTerm]?.loaded)
-        ? m(List, [ m(emptySearchPreview, { searchTerm }) ])
-        : LoadingPreview
-      : vnode.state.isTyping
-        ? LoadingPreview
-        : m(List, { class: 'search-results-list' }, results);
-    vnode.state.closeResults = () => { vnode.state.hideResults = true; };
+    const scopeButtons = [SearchScope.Threads, SearchScope.Replies].map(s => scopeToButton(s, false))
```

Comment:
Might be better to remove unavailable options entirely, rather than graying them out. Saves some vertical space, perhaps less confusing? @zakhap thoughts? 
![image](https://user-images.githubusercontent.com/22281010/146815097-ba94babe-7329-48ab-9469-7533e36712bf.png)

Reply:
These toggle once you unfilter by chain, so I think they make most sense visible but disabled, rather than popping in and out of existence as you toggle by chain. Open to input though!

Reply:
I agree with Liam in this case! It looks good to me, but when we have real users actually use it, we'll get more feedback in the upcoming weeks for revision.